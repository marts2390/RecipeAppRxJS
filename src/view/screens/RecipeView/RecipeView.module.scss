@import "src/view/styles/variables.scss";

.recipeView {
  padding: 0.5em;
  max-width: 1600px;
  margin: 0 auto;

  @media (min-width: $screen-md) {
    padding: 2em;
  }
}

.spinnerContainer {
  height: 100%;
  width: 100%;
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
}
