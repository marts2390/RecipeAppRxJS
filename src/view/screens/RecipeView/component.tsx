import React, { PureComponent } from 'react'

import Loader from 'view/components/Loader'
import RecipeViewHeader from './RecipeViewHeader'
import RecipeViewIngredients from './RecipeViewIngredients'

import styles from './RecipeView.module.scss'

interface IRecipeViewComponentProps {
  getRecipeData: (recipeId: string) => void;
  resetState: () => void;
  recipeId: string;
  isLoaded: boolean;
}

class RecipeViewComponent extends PureComponent<IRecipeViewComponentProps> {
  componentDidMount() {
    this.getData()
  }

  componentWillUnmount() {
    const { resetState } = this.props

    resetState()
  }

  getData = () => {
    const { getRecipeData, recipeId } = this.props

    if (recipeId) {
      getRecipeData(recipeId)
    }
  }

  render() {
    const { isLoaded } = this.props
    return (
      <div>
        {isLoaded ? (
          <div className={ styles.recipeView }>
            <RecipeViewHeader />
            <RecipeViewIngredients />
          </div>
        ) : (
          <div className={ styles.spinnerContainer }>
            <Loader />
          </div>
        )}
      </div>
    )
  }
}

export default RecipeViewComponent
