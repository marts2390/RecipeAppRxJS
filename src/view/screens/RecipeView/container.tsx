import React, { FC } from 'react'
import UC from 'actions'
import { useObservable } from 'rxjs-hooks'

import { Match } from 'models/reactRouterDomModel'

import RecipeViewComponent from './component'

interface IRouterProps {
  recipeId: string;
}

interface IRecipeViewContainerProps {
  match: Match<IRouterProps>;
}

const RecipeViewContainer:FC<IRecipeViewContainerProps> = ({ match }) => {
  const { isLoaded } = useObservable(UC.RecipeScreenService.getSubject) || UC.RecipeScreenService.getCurrentState()
  const { recipeId } = match.params

  return (
    <RecipeViewComponent
      getRecipeData={ UC.RecipeScreenService.getRecipeData }
      resetState={ UC.RecipeScreenService.setDefaultState }
      recipeId={ recipeId }
      isLoaded={ isLoaded }
    />
  )
}

export default RecipeViewContainer
