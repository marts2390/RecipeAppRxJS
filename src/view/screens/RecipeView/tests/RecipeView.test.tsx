import React from 'react'
import { shallow } from 'enzyme'
import RecipeView from '../component'

let wrapper
let props

beforeEach(() => {
  props = {
    getRecipeData: jest.fn(),
    resetState: jest.fn(),
    recipeId: '',
    isLoaded: false,
  }

  wrapper = shallow(<RecipeView { ...props } />)
})

test('Recipe View is rendered with data', () => {
  wrapper.setProps({ isLoaded: true })
  expect(wrapper).toMatchSnapshot()
})

test('Should display loader', () => {
  expect(wrapper).toMatchSnapshot()
})

test('get initial load', () => {
  const instance = wrapper.instance()
  jest.spyOn(instance, 'getData')
  instance.componentDidMount()
  expect(instance.getData).toHaveBeenCalledTimes(1)
})
