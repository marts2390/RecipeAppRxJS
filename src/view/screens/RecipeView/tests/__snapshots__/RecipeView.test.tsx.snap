// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe View is rendered with data 1`] = `
<div>
  <div
    className="recipeView"
  >
    <RecipeViewHeaderContainer />
    <RecipeViewIngredientsContainer />
  </div>
</div>
`;

exports[`Should display loader 1`] = `
<div>
  <div
    className="spinnerContainer"
  >
    <LoaderComponent />
  </div>
</div>
`;
